code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -724,6 +730,7 @@
 
 	if(dead_xeno == living_xeno_ruler)
 		on_ruler_death(dead_xeno)
+		dead_xeno.remove_ruler_abilities()
```

Comment:
doesnt update ruler change this? i feel like this or onruler death should be handling it already